Natalie Meintjes Capstone project

natalie-mein opened the pull request:
### Data Engineer Capstone Project Submission

**Contributor Name**:  
**Project Title**:  
**Short Description**:

### Checklist

- [ ] I created a folder with my name
- [ ] My project includes a `README.md`
- [ ] All files are inside my folder
- [ ] I tested the DAGs/scripts locally
- [ ] I am open to feedback and discussion

chanukyapekala commented:
Thanks for your submission. Great work and modularized structure. 

Few comments :

1. Any thought process of what you would need to do if volumes grow enormously?
2. Who is the end user of this analysis datasets, would it be a dashboard or API or some model - this part is completely missing. 
3. What if you package this code and share with others as a wheel file? Its more standard and secure that way than just filename.py
4. Unit Tests if any for your functions or API calls?
5. An architecture diagram by simply using draw.io
6. commit message should reflect the state of code, can be improved there! 

No pressure, just if you are interested extend the capstone project to a scalable package as well. 

Let me know what you think

chanukyapekala commented:
Have you finalized the changes?

natalie-mein commented:
Hi,

Thanks again for the feedback. I'll respond with some thoughts here:

**1. Any thought process of what you would need to do if volumes grow enormously?**
If volumes were to grow there are a few approaches I could take. For instance, switching to chunked reading or distributed tools like Dask for cleaning and transformation. For storage, Parquet might be a better option than CSV. If it gets very big then parts of the project could  be isolated in different notebooks. I think Airflow could also be helpful in the case of very large data sets, but I haven't got any experience with it, and I can't get it to run on my computer as of yet. Hopefully I'll have a chance to experiment with it in the future.

**2. Who is the end user of this analysis datasets, would it be a dashboard or API or some model - this part is completely missing.**
This is a great point. As this was my first attempt at a data engineering project, i focused mostly on ingesting, cleaning and analyzing the data in Jupyter notebooks. I didn't define specific end user or output format like dashboard or API, but that is something I would like to explore in the future. I will revisit this once I know more about APIs etc.

Right now, the output is mostly visual (plots and summary statistics), intended for human interpretation, but it would be very useful to make the data available in a dashboard or model. 

**3. What if you package this code and share with others as a wheel file? Its more standard and secure that way than just filename.py**
This is a great suggestion! I was not familiar with wheel files before but in the bit of research I've done it does seem like a better option. In future versions I can turn this into a proper python package.

**4. Unit Tests if any for your functions or API calls?**
I added a unit test for the clean_data.py file.

**5. An architecture diagram by simply using draw.io**
I added a simple diagram from draw.io.

**6. commit message should reflect the state of code, can be improved there!**
I actually had very informative commit messages but my commits failed so many times (because of my large csv file which I ended up being unable to include as LFS cannot be used on forked directories. Who knew?!) , I didn't actually know which one would work and they ended up being kind of rubbish. But I will be more mindful of this in the future.

chanukyapekala commented:
Alright, Thanks for your comments. I am approving the PR. Please use option Squash and Merge in the pull request.
